### PR TITLE
feat: [PL-41388]: Added public endpoints for Harness OIDC provider

### PR DESCRIPTION
**Pull request by tiwaritejas**
## Describe your changes
This PR adds the following endpoints for Harness OIDC provider as per the Open ID Connect spec-
1. .wellknown/openid-configuration
2. .well-known/jwks
## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/54084)

### CODE REVIEW

File: 120-ng-manager/src/main/java/io/harness/ng/core/oidc/OidcResource.java [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+      throw new InternalServerErrorException("Failed due to internal server error. Please try again after some time.");
+    }
+    String issuer = String.format("%s/account/%s", baseUrl, accountId);
+    JSONObject openIdConfig = new JSONObject();
+    openIdConfig.put("issuer", issuer);
+    openIdConfig.put("jwks_uri", String.format("%s/.well-known/jwks", issuer));
+    openIdConfig.put("subject_types_supported", List.of("public", "pairwise"));
+    openIdConfig.put("response_types_supported", List.of("id_token"));
+    openIdConfig.put(
+        "claims_supported", List.of("sub", "aud", "exp", "iat", "iss", "pipeline_name", "pipeline_exec_attempt"));
```
Comment (Raghav-Murali) [inline 120-ng-manager/src/main/java/io/harness/ng/core/oidc/OidcResource.java:113]: Only these - "sub","aud","exp","iat","iss","account_id"

File: 120-ng-manager/src/main/java/io/harness/ng/core/oidc/OidcResource.java [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      , @Content(mediaType = "application/yaml", schema = @Schema(implementation = FailureDTO.class))
+    })
+@io.swagger.v3.oas.annotations.responses.ApiResponse(responseCode = "500", description = "Internal server error",
+    content =
+    {
+      @Content(mediaType = "application/json", schema = @Schema(implementation = ErrorDTO.class))
+      , @Content(mediaType = "application/yaml", schema = @Schema(implementation = ErrorDTO.class))
+    })
+@AllArgsConstructor(onConstructor = @__({ @Inject }))
+@NextGenManagerAuth
```
Comment (Raghav-Murali) [inline 120-ng-manager/src/main/java/io/harness/ng/core/oidc/OidcResource.java:75]: Remove this as discussed.

**Comment by tiwaritejas** [comment]
trigger uts

**Comment by Raghav-Murali** [comment]
trigger uts

**Comment by tiwaritejas** [comment]
trigger uts

**Review by sahibaHarness — APPROVED** [review]

**Pull request merged**